File: VB/GenerateUserFriendlyId.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports GenerateUserFriendlyId.Module.Utils

Namespace GenerateUserFriendlyId.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As ObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			SequenceGenerator.RegisterSequences(XafTypesInfo.Instance.PersistentTypes)
		End Sub
	End Class
End Namespace

File: CS/CS/GenerateUserFriendlyId.Module/Updater.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using GenerateUserFriendlyId.Module.Utils;

namespace GenerateUserFriendlyId.Module {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            //Dennis: It is necessary to manually initialize SequenceGenerator when database is updated via the DBUpdater tool.
            SequenceGeneratorInitializer.Initialize();
            //Dennis: It is necessary to register sequences for persistent types used in your application.
            SequenceGenerator.RegisterSequences(XafTypesInfo.Instance.PersistentTypes);
        }
    }
}